### PR TITLE
Manually depad RSAES-PKCS1 on Apple OSes

### PR DESCRIPTION
**Pull request by bartonjs**
This provides a workaround for a behavioral change in the latest macOS update, to keep the platform behaving consistently (and tests passing on fully patched machines).

### CODE REVIEW

**Review by vcsjones — COMMENTED** [review]
Jeremy and I have been talking about this one in another medium. But to translate over the commentary:

1. There is a lack of test coverage for this change. The existing rsaes-pkcs1-v1_5 tests are lacking.
2. Preliminary tests uncovered an issue.

File: src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.RSA.cs [hunk 1]
```diff
@@ -69,8 +70,8 @@ internal static partial class AppleCrypto
             out SafeCFDataHandle pEncryptedOut,
             out SafeCFErrorHandle pErrorOut);
 
-        [LibraryImport(Libraries.AppleCryptoNative, EntryPoint = "AppleCryptoNative_RsaDecryptPkcs")]
-        private static partial int RsaDecryptPkcs(
+        [LibraryImport(Libraries.AppleCryptoNative, EntryPoint = "AppleCryptoNative_RsaDecryptRaw")]
```
Comment (vcsjones) [inline src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.RSA.cs:73]: `AppleCryptoNative_RsaDecryptPkcs` is effectively dead code at this point. We can remove the native code and associated `DllImportEntry`, but if you want to hang on to it for whatever reason, that's fine.

File: src/libraries/Common/src/System/Security/Cryptography/RsaPaddingProcessor.cs [hunk 2]
```diff
@@ -142,6 +143,109 @@ internal static int HashLength(HashAlgorithmName hashAlgorithmName)
...
+            Span<byte> destination,
+            out int bytesWritten)
+        {
+            int primitive = DepadPkcs1Encryption(source);
+            int primitiveSign = SignStretch(primitive);
+
+            // Primitive is a positive length, or ~length to indicate
+            // an error, so flip ~length to length if the high bit is set.
+            int len = Choose(primitiveSign, ~primitive, primitive);
+            int spaceRemain = destination.Length - len;
```
Comment (vcsjones) [inline src/libraries/Common/src/System/Security/Cryptography/RsaPaddingProcessor.cs:157]: Can this overflow from subtracting a large negative?

Comment (bartonjs) [reply src/libraries/Common/src/System/Security/Cryptography/RsaPaddingProcessor.cs:157]: Right.  `len` is always in `[0, input.Length - 11]`

**Review by vcsjones — APPROVED** [review]

File: src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.RSA.cs (hunk 1 above)
Comment (bartonjs) [reply src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.RSA.cs:73]: My hope is that we can jump back to it at some point, so I was leaving it easy.

**Pull request merged**